Sources/BraintreePayPal/BTPayPalClient.swift

```diff
@@ -439,7 +439,11 @@ import BraintreeDataCollector
 
             switch returnURL.state {
             case .succeeded, .canceled:
-                handleReturn(url, paymentType: .vault, completion: completion)
+                guard let payPalRequest else {
+                    notifyFailure(with: BTPayPalError.missingPayPalRequest, completion: appSwitchCompletion)
+                    return
+                }
+                handleReturn(url, paymentType: payPalRequest.paymentType, completion: completion)
```

Comment:
Do we want to make this same change here or leave that section hardcoded since App Switch is vault only currently: https://github.com/braintree/braintree_ios/blob/5a8e02f03c898a5e180955ba47668d2c9db8eaa6/Sources/BraintreePayPal/BTPayPalClient.swift#L294